analysis/runtime/src/runtime/backend.rs

```diff
@@ -0,0 +1,95 @@
+use enum_dispatch::enum_dispatch;
+use fs_err::File;
+use std::env;
+use std::io::BufWriter;
+use std::sync::mpsc::Receiver;
+
+use bincode;
+
+use super::{AnyError, FINISHED};
+use crate::events::{Event, EventKind};
+use crate::metadata::Metadata;
+
+#[enum_dispatch]
+pub(super) trait IBackend {
+    fn write(&mut self, event: Event);
+}
+
+pub struct DebugBackend {
+    metadata: Metadata,
+}
+
+impl IBackend for DebugBackend {
+    fn write(&mut self, event: Event) {
+        let mir_loc = self.metadata.get(event.mir_loc);
+        eprintln!("{:?}: {:?}", mir_loc, event.kind);
+    }
+}
+
+pub struct LogBackend {
+    writer: BufWriter<File>,
+}
+
+impl IBackend for LogBackend {
+    fn write(&mut self, event: Event) {
+        bincode::serialize_into(&mut self.writer, &event).unwrap();
+    }
+}
+
+#[enum_dispatch(IBackend)]
+pub enum Backend {
```

Comment:
This would then be `BackendKind` or such.

Reply:
`BackendKind` doesn't make sense since the `enum` has data.  If it didn't and were just the kind, then it'd make sense.  Also then `{Log,Debug}Backend` should be `Kind`s, too, but that makes even less sense.

Reply:
Yeah, I realized this right after commenting, but the `IBackend` name is still not ideal. Maybe `WriteEvent` would be a better name for the trait, as that's the only method exposed.

Reply:
Yeah, that's a better idea.  Should it be `WriteEvent` or `EventWriter`?

Reply:
The former--the more rusty convention is imperative, like `Read`/`Write`/`Debug`/`Serialize`.

Reply:
Ah yes, okay, I'll rename that and then merge.

Reply:
Fixed in https://github.com/immunant/c2rust/pull/490/commits/e3098a14131f0acd39db71a5dbc3d0709fcb694f.